### ISSUE TITLE
Masked array with object dtype doesn't unpickle

### ISSUE THREAD

**Issue by amueller** [numpy/numpy#8113]
```python
import numpy as np
import pickle
asdf = np.ma.MaskedArray([1, 2, 3], dtype=object)
pickle.loads(pickle.dumps(asdf))
```
> TypeError: object pickle not returning list

NumPy 1.11.1, python 3.5.2

**Comment by jnothman** [comment]
Related: #1495

**Comment by raghavrv** [comment]
Could be closed as #8122 is merged

**Comment by shoyer** [comment]
@raghavrv indeed, done
